File: JS/Tree/226.Invert Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    if (root) {
        const temp = invertTree(root.left);
        root.left = invertTree(root.right);
        root.right = temp;
    }
    return root;
};

var invertTree = function(root) {
    const stack = [root];
    while(stack.length) {
        const node = stack.pop();
        if (node) {
            const temp = node.left;
            node.left = node.right;
            node.right = temp;
            stack.push(node.left);
            stack.push(node.right);
        }
    }
    return root;
};

File: Golang/Tree/226.invert-binary-tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	var recursive func(node *TreeNode)
	recursive = func(node *TreeNode) {
		if node != nil {
			recursive(node.Left)
			recursive(node.Right)
			node.Left, node.Right = node.Right, node.Left
		}
	}
	recursive(root)
	return root
}
